src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -422,6 +413,11 @@ public synchronized void close() {
                 }
             }
 
+            for (PreparedStatementCapsule ps : stmts.values()) {
+                ps.ps.close();
```

Comment:
Consider wrapping in a try/catch so that we try to close as many as possible (and avoid misleading the user with a log message stating "Unable to close connection"

Reply:
Also, since you're here, could you please add the exception instance to the log message in line 424?

Reply:
Addressed in b9c7fc0